### PR TITLE
fixed Java method overload bug,

### PR DESCRIPTION
**Pull request by laiweiwei**
e.g. 
1. Java: 
```
method(JsonArray array); 
method(JsonObject object);
```
2. JS:
```
java.method([ "a", "b", "c" ]);// JS Native Array can convert to Java Map, { 0 : "a", 1 : "b", 2 : "c" },
java.method({ name:"vv", age:18 })
```

### CODE REVIEW

**Comment by laiweiwei** [comment]
before:
```
// map native JSON Object to Vert.x JSONObject
.targetTypeMapping(Map.class, JsonObject.class, null, JsonObject::new)
```
after:
```
.targetTypeMapping(Map.class, JsonObject.class, v -> !Value.asValue(v).hasArrayElements(), JsonObject::new)
```

**Comment by laiweiwei** [comment]
whithout `v -> !Value.asValue(v).hasArrayElements()` the GraalJS will throws an error:
```
TypeError: execute on foreign object failed due to: Multiple applicable overloads found for method name 
```

**Comment by laiweiwei** [comment]
https://github.com/graalvm/graaljs/issues/3

File: es4x/src/main/java/io/reactiverse/es4x/ECMAEngine.java [hunk 1]
```diff
@@ -119,7 +119,7 @@ public ECMAEngine(Vertx vertx) {
       // Ensure bytes are supported
       .targetTypeMapping(Number.class, Byte.class, null, Number::byteValue)
       // map native JSON Object to Vert.x JSONObject
-      .targetTypeMapping(Map.class, JsonObject.class, null, JsonObject::new)
+      .targetTypeMapping(Map.class, JsonObject.class, v -> !Value.asValue(v).hasArrayElements(), JsonObject::new)
       // map native JSON Array to Vert.x JSONArray
       .targetTypeMapping(List.class, JsonArray.class, null, JsonArray::new)
```
Comment (pmlopes) [inline es4x/src/main/java/io/reactiverse/es4x/ECMAEngine.java:124]: Following the same logic, this `null` should be the reversed assertion then?

**Comment by laiweiwei** [comment]
> Should we also have the reserve assertion for lists then?

No need to do that i think.

**Comment by laiweiwei** [comment]
```
.targetTypeMapping(Map.class, JsonObject.class, null, JsonObject::new)
.targetTypeMapping(List.class, JsonArray.class, null, JsonArray::new)
```
Both of these 2 conditions are acceptable to the `JS Native Array`, hherefore, for GraalJS, multiple choices are confused, the solution is to add a clear filter condition: 
```
.targetTypeMapping(Map.class, JsonObject.class, !Value.asValue(v).hasArrayElements(), JsonObject::new)
.targetTypeMapping(List.class, JsonArray.class, null, JsonArray::new)
```
For `Map.class`, it will match both an `[]` and an `{}`,
But for `List.class`, it does not match both an `[]` and an `{}`, 
so i dont think we should also have the reserve assertion for List.cass case.

**Pull request merged**